tests/rptest/tests/schema_registry_test.py

```diff
@@ -1073,6 +1073,17 @@ def test_post_subjects_subject_versions(self):
         result = result_raw.json()
         # assert result["schema"] == json.dumps(schema_def)
 
+        self.logger.debug(
+            "Post schema 1 with escape chars in the subject name")
+        name = f"{topic}%2fkey"
+        result_raw = self._post_subjects_subject_versions(subject=name,
+                                                          data=schema_1_data)
+        self.logger.warn(result_raw)
+        assert result_raw.status_code == requests.codes.ok
+        result_raw = self._get_subjects_subject_versions(subject=name)
+        assert result_raw.status_code == requests.codes.ok
+        assert result_raw.json() == [1]
```

Comment:
It's not immediately obvious to me that a post followed by a get wouldn't suffer some kind of symmetric encode/decode problem. Indeed, this test does not fail with the old parsing.

Something like this would instil me with more confidence:

```suggestion
        result_raw = self._get_subjects()
        assert result_raw.status_code == requests.codes.ok
        assert name in result_raw.json(), f"{name} not in {result_raw.content}"
```
But that test still fails.

Reply:
Yeah good catch. The issue was with the test case, which should have had two "levels" of URL encoding to exercise bug. Update fails against the old code, passes here.